runtime/service.go

```diff
@@ -681,6 +668,33 @@ func (s *service) buildVMConfiguration(req *proto.CreateVMRequest) (*firecracker
 	return &cfg, nil
 }
 
+func (s *service) buildNonStubDrives(req *proto.CreateVMRequest) []models.Drive {
+	var builder firecracker.DrivesBuilder
+
+	if input := req.RootDrive; input != nil {
+		builder = firecracker.DrivesBuilder{}.
+			WithRootDrive(input.PathOnHost,
+				firecracker.WithPartuuid(input.Partuuid),
+				firecracker.WithReadOnly(!input.IsWritable),
+				func(d *models.Drive) {
+					if limiter := input.RateLimiter; limiter != nil {
+						d.RateLimiter = rateLimiterFromProto(limiter)
+					}
+				})
+	} else {
+		builder = firecracker.DrivesBuilder{}.
+			WithRootDrive(s.config.RootDrive, func(d *models.Drive) {
```

Comment:
`WithReadOnly` can be used here

Reply:
Good catch! Addressed at https://github.com/firecracker-microvm/firecracker-containerd/pull/283/commits/96c7417c7fe06a4d8f2c25150c83010d9e6e400c.